test/access-token.test.js

```diff
@@ -174,30 +173,83 @@ describe('loopback.token(options)', function() {
           assert(!err);
-          assert.deepEqual(res.body, {userId: userId, state: 1});
+          assert.deepEqual(res.body, { userId: userId, state: 1 });
           done();
         });
     });
 
-  it('should skip when req.token is already present', function(done) {
-    var tokenStub = { id: 'stub id' };
-    app.use(function(req, res, next) {
-      req.accessToken = tokenStub;
-      next();
+  describe('loading multiple instances of token middleware', function() {
+    it('should skip when req.token is already present and no further options are set', function(done) {
+      var tokenStub = { id: 'stub id' };
+      app.use(function(req, res, next) {
+        req.accessToken = tokenStub;
+        next();
+      });
+      app.use(loopback.token({ model: Token }));
+      app.get('/', function(req, res, next) {
+        res.send(req.accessToken);
+      });
+
+      request(app).get('/')
+        .set('Authorization', this.token.id)
+        .expect(200)
+        .end(function(err, res) {
+          if (err) return done(err);
+          expect(res.body).to.eql(tokenStub);
+          done();
+        });
     });
-    app.use(loopback.token({ model: Token }));
-    app.get('/', function(req, res, next) {
-      res.send(req.accessToken);
+
+    it('should not overwrite valid existing token (has "id" property) when overwriteExistingToken is falsy', function(done) {
+      var tokenStub = { id: 'stub id' };
+      app.use(function(req, res, next) {
+        req.accessToken = tokenStub;
+        next();
+      });
+      app.use(loopback.token({
+        model: Token,
+        enableDoublecheck: true,
+      }));
+      app.get('/', function(req, res, next) {
+        res.send(req.accessToken);
+      });
+
+      request(app).get('/')
+        .set('Authorization', this.token.id)
+        .expect(200)
+        .end(function(err, res) {
+          if (err) return done(err);
+          expect(res.body).to.eql(tokenStub);
+          done();
+        });
     });
 
-    request(app).get('/')
-      .set('Authorization', this.token.id)
-      .expect(200)
-      .end(function(err, res) {
-        if (err) return done(err);
-        expect(res.body).to.eql(tokenStub);
-        done();
+    it('should overwrite existing token when enableDoublecheck and overwriteExistingToken options are truthy', function(done) {
+      var token = this.token;
+      var tokenStub = { id: 'stub id' };
+
+      app.use(function(req, res, next) {
+        req.accessToken = tokenStub;
+        next();
+      });
+      app.use(loopback.token({
+        model: Token,
+        enableDoublecheck: true,
+        overwriteExistingToken: true
+      }));
+      app.get('/', function(req, res, next) {
+        res.send(req.accessToken);
       });
+
+      request(app).get('/')
+        .set('Authorization', token.id)
+        .expect(200)
+        .end(function(err, res) {
+          if (err) return done(err);
+          expect(res.body.userId).to.eql(token.userId);
```

Comment:
Why can't we do `expect(res.body).to.eql(token)` here, as we already do in other test cases? What am I missing?